superset-frontend/src/explore/components/ExploreViewContainer/index.jsx

```diff
@@ -699,12 +699,17 @@ function mapStateToProps(state) {
   const chartKey = Object.keys(charts)[0];
   const chart = charts[chartKey];
 
+  let dashboardId = Number(explore.form_data?.dashboardId);
+  if (Number.isNaN(dashboardId)) {
```

Comment:
`??` operator doesn't exclude NaNs. I also don't want to use `||` operator in order not to exclude dashboards with id `0`